src/playground-service-worker-proxy.ts

```diff
@@ -46,6 +51,19 @@ import {
     {scope}
   );
 
+  window.addEventListener(
+    'message',
+    (event: MessageEvent<PlaygroundMessage>) => {
+      if (event.data.type === UPDATE_SERVICE_WORKER) {
+        // When the project handshakes with the service worker, it may notice a
+        // version mismatch, in which case it will send this message to request
+        // the service worker update. Note that service workers eventually
+        // update automatically, but not necessarily right away.
+        registration.update();
```

Comment:
I'm not able to follow where/when this update causes `CONNECT_PROJECT_TO_SW` to be called again, resulting in [this check](https://github.com/PolymerLabs/playground-elements/pull/178/files#diff-8f05853e3ce4da716aed3677ad38b34cf6423aaf403286a5a8ae458b97a85f5cR436) running again with same version.

My guess from my limited understanding is that this update triggers a `message` event here: https://github.com/PolymerLabs/playground-elements/pull/178/files#diff-8f05853e3ce4da716aed3677ad38b34cf6423aaf403286a5a8ae458b97a85f5cR457 .

Reply:
Calling update forces the update lifecycle to begin immediately, as opposed to at some indeterminate/opportunistic time in the future. That lifecycle is described here: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API#download_install_and_activate

TL;DR The service worker script will be fetched, and byte-compared to the current one. If it's different, then the service worker gets an `activate` + `install` event, and then our registration object here will get an `updatefound` event, and then we call our `connect` function here to do a new handshake with the project element.

Expanded the comment here.